great_expectations/rule_based_profiler/data_assistant/onboarding_data_assistant.py

```diff
@@ -614,7 +615,7 @@ def _build_datetime_columns_rule() -> Rule:
 
         # Step-1: Instantiate "ColumnDomainBuilder" for selecting proper datetime columns (not "datetime-looking" text).
 
-        datetime_column_type_domain_builder = ColumnDomainBuilder(
+        datetime_column_type_domain_builder: DomainBuilder = ColumnDomainBuilder(
```

Comment:
I don't think we need type hints here - the type is inferred from the right-hand side of the expression and the relation to the parent is implicit

Reply:
Agreed

Reply:
@cdkini This is to emphasize the polymorphism.  Thanks.